eodag/utils/constraints.py

```diff
@@ -0,0 +1,223 @@
+            )
+
+    return queryables
+
+
+def fetch_constraints(
+    constraints_url: str, plugin: Union[Search, Api]
+) -> List[Dict[Any, Any]]:
+    """
+    fetches the constraints from a provider
+    :param constraints_url: url from which the constraints can be fetched
+    :type constraints_url: str
+    :param plugin: api or search plugin of the provider
+    :type plugin: Union[Search, Api]
+    :returns: list of constraints fetched from the provider
+    :rtype: List[Dict[Any, Any]]
+    """
+    try:
+        headers = USER_AGENT
+        logger.debug("fetching constraints from %s", constraints_url)
+        if hasattr(plugin, "auth"):
+            res = requests.get(
+                constraints_url,
+                headers=headers,
+                auth=plugin.auth,
+                timeout=HTTP_REQ_TIMEOUT,
+            )
+        else:
+            res = requests.get(
+                constraints_url, headers=headers, timeout=HTTP_REQ_TIMEOUT
+            )
+        res.raise_for_status()
+    except requests.exceptions.Timeout as exc:
+        raise TimeOutError(exc, timeout=HTTP_REQ_TIMEOUT) from exc
+    except requests.exceptions.HTTPError as err:
+        logger.error(
+            "constraints could not be fetched from %s, error: %s",
+            constraints_url,
+            str(err),
+        )
+        return []
+    else:
+        constraints_data = res.json()
+        config = plugin.config.__dict__
+        if (
+            "constraints_entry" in config
+            and config["constraints_entry"]
+            and config["constraints_entry"] in constraints_data
+        ):
+            constraints = constraints_data[config["constraints_entry"]]
+        else:
+            constraints = constraints_data
+        return constraints
+
+
+def _get_provider_queryable_key(
+    eodag_key: str,
+    provider_queryables: Dict[str, Any],
+    plugin: Union[Search, Api],
+    product_type: Optional[str] = None,
+) -> str:
+    """finds the provider queryable corresponding to the given eodag key based on the metadata mapping
+    :param eodag_key: key in eodag
+    :type eodag_key: str
+    :param provider_queryables: queryables returned from the provider
+    :type provider_queryables: dict
+    :param plugin: plugin from which the config is taken
+    :type plugin: Union[Api, Search]
+    :param product_type: product type for which the metadata should be used
+    :type product_type: Optional[str]
+    :returns: provider queryable key
+    :rtype: str
+    """
+    metadata_mapping = _get_metadata_mapping(plugin, product_type)
+    if eodag_key not in metadata_mapping:
+        return ""
+    mapping_key = metadata_mapping[eodag_key]
+    if isinstance(mapping_key, list):
+        for queryable in provider_queryables:
+            if queryable in mapping_key[0]:
+                return queryable
+        return ""
+    else:
+        return eodag_key
+
+
+def _get_metadata_mapping(
```

Comment:
product-type specific metadata-mapping merge should be done in `search_plugin.__init__`, see `QueryStringSearch.__init__`